### PR TITLE
Add Mock for Mock Smm Sw Dispatch2 Protocol

### PR DESCRIPTION
**Pull request by Rasheed-Yusuf**
## Description

Add Mock for Mock Smm Sw Dispatch2 Protocol

For details on how to complete these options and their meaning refer to [CONTRIBUTING.md](https://github.com/microsoft/mu/blob/HEAD/CONTRIBUTING.md).

- [ ] Impacts functionality?
- [ ] Impacts security?
- [ ] Breaking change?
- [ ] Includes tests?
- [ ] Includes documentation?
- [x] Backport to release branch?

## How This Was Tested

Incorporated into GoogleTest

## Integration Instructions

N/A

### CODE REVIEW

File: MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockFirmwareVolumeBlock.h [hunk 1]
```diff
@@ -0,0 +1,103 @@
+/** @file MockFirmwareVolumeBlock.h
+  This file declares a mock of Firmware Volume Block Protocol
+
+  Copyright (c) Microsoft Corporation.
+  Your use of this software is governed by the terms of the Microsoft agreement under which you obtained the software.
```
Comment (makubacki) [inline MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockFirmwareVolumeBlock.h]: This should be replaced with the SPDX license identifier in all files.

```suggestion
  SPDX-License-Identifier: BSD-2-Clause-Patent
```

File: MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockSmmSwDispatch2.h [hunk 2]
```diff
@@ -0,0 +1,56 @@
+/** @file MockSmmSwDispatch2.h
+  This file declares a mock of SMM Software Dispatch Protocol
+
+  Copyright (c) Microsoft Corporation.
+  Your use of this software is governed by the terms of the Microsoft agreement under which you obtained the software.
```
Comment (makubacki) [inline MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockSmmSwDispatch2.h]:
```suggestion
  SPDX-License-Identifier: BSD-2-Clause-Patent
```

File: MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockFirmwareVolumeBlock.h
Comment (makubacki) [inline MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockFirmwareVolumeBlock.h:1]: Did you intend to add the firmware volume block mock to both this PR and https://github.com/microsoft/mu_basecore/pull/1216?

Comment (Rasheed-Yusuf) [reply MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockFirmwareVolumeBlock.h:1]: No. That was unintended.

**Review by apop5 — APPROVED** [review]

File: MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockSmmSwDispatch2.h [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+     IN CONST EFI_SMM_SW_DISPATCH2_PROTOCOL  *This,
+     IN       EFI_HANDLE                     DispatchHandle)
+    );
+};
+
+MOCK_INTERFACE_DEFINITION (MockSmmSwDispatch2Protocol);
+MOCK_FUNCTION_DEFINITION (MockSmmSwDispatch2Protocol, MockRegister, 4, EFIAPI);
+MOCK_FUNCTION_DEFINITION (MockSmmSwDispatch2Protocol, MockUnRegister, 2, EFIAPI);
+
+EFI_SMM_SW_DISPATCH2_PROTOCOL  SMM_SW_DISPATCH2_PROTOCOL_MOCK = {
```
Comment (VivianNK) [inline MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockSmmSwDispatch2.h]: Based on the latest discussions in Edk2, the definition of MockEfiSmmSwDispatch2Protocol  should be a #define. Please see the example below:

https://github.com/tianocore/edk2/pull/5950/files#diff-bce28afad342dcfbadd547d3d185dc2159e235b39e204990b683711c22dfc5e9

File: MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockSmmSwDispatch2.h [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+#include <Library/GoogleTestLib.h>
+#include <Library/FunctionMockLib.h>
+
+extern "C" {
+  #include <Uefi.h>
+  #include <Protocol/SmmSwDispatch2.h>
+}
+
+// Declarations to handle usage of the EFI_SMM_SW_DISPATCH2_PROTOCOL
+struct MockSmmSwDispatch2Protocol {
```
Comment (VivianNK) [inline MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockSmmSwDispatch2.h]:
```suggestion
struct MockEfiSmmSwDispatch2Protocol {

```

**Review by VivianNK — APPROVED** [review]

**Pull request merged**